tools/setup_dev_environment.py

```diff
@@ -120,6 +132,8 @@ def main():
             )
             if p.returncode != 0:
                 print(f"Failed to install {latest}")
+        pin_version(latest, "pip", PIP_VERSION)
+        pin_version(latest, "tox", TOX_VERSION)
```

Comment:
could you remove the duplication on lines 151-157, when we install tox again?

Reply:
So that wasn't actually redundant in the sense that the one I added here is inside the virtual environment and only matters if that env is activated; generally `tox` is going to be run from outside and the original one will matter. I refactored to combine them into the same function though.

Reply:
sounds good